### PR TITLE
fix: fixed download_sample_videos.sh should support non dgpu systems

### PR DESCRIPTION
**Pull request by vli11**
Closes: issue #38

_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
_Provide a 1-2 line brief overview of the changes submitted through the Pull Request..._

### REFERENCES
Reference URL for issue tracking (Github): **\<URL to be filled>**

- [ ] Added label to the Pull Request for easier discoverability and search

### CODE MAINTAINABILITY
- [x] **_Commit Message meets guidelines as indicated in the URL\*_**
	- https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [ ] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [ ] Updated Documentation as relevant to the changes
- [ ] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approve the Pull Request for merge are required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check.

### QUALITY CHECKS
- [ ] Architectural and Design Fit
- [ ] **_Quality of code (At least one should be checked as applicable)\*_**
	- [ ] Commit Message meets guidelines
	- [ ] PR changes adhere to industry practices and standards
	- [ ] Adopted domain specific coding standards ([SBL Coding Standards](https://github.com/tianocore-docs/Docs/raw/master/Specifications/CCS_2_1_Draft.pdf))
	- [ ] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [ ] Code is adequately commented
	- [ ] Code copyright is correct
	- [ ] Tracing output are minimized and logic
	- [ ] Confusing logic is explained in comments
	- [ ] Commit comment can be used to design a new test case for the changes
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

# _Code must act as a teacher for future developers_

### CODE REVIEW

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
- line 23: `source ../get-gpu-info.sh`  not working for me if i ran just this script.    The other script `get-gpu-info.sh` is on the same directory as this script.

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
would also suggest writing a unit test shell script to test this script

File: benchmark-scripts/format_avc_mp4.sh [hunk 1]
```diff
@@ -70,14 +49,33 @@ then
...
 
 if [ $HAS_FLEX_140 == 1 ] || [ $HAS_FLEX_170 == 1 ] || [ $HAS_ARC == 1 ]
 then
         TAG=sco-dgpu:2.0
 
 else
-        echo "ERROR: Requires Intel Flex/Arc GPU"
-	exit 1
+        TAG=sco-soc:2.0
 fi
```
Comment (jim-wang-intel) [inline benchmark-scripts/format_avc_mp4.sh:79]: @brian-intel :   if either one of docker images can work, why do we need this part of logic at all? 
we could just use either one of images and then we don't even need the logic to detect GPU for downloading media file,  right?

**Review by jim-wang-intel — COMMENTED** [review]
question for gpu logic

File: benchmark-scripts/format_avc_mp4.sh (hunk 1 above)
Comment (brian-intel) [reply benchmark-scripts/format_avc_mp4.sh:79]: The main issue is most users will only build the image based on their platform. We could possibly change the logic to use whichever image is available on the system.

Comment (jim-wang-intel) [reply benchmark-scripts/format_avc_mp4.sh:79]: yea that's true: so if we could do grep docker images then we can get rid of gpu detection logic then -- i like that idea

File: benchmark-scripts/format_avc_mp4.sh [hunk 2]
```diff
@@ -20,66 +20,79 @@ WIDTH=3840
...
-	WIDTH=$3
+	echo "ERROR: Invalid height input"
+	exit 1
 fi
 
-if [ ! -z "$4" ]
+if ! [[ "$FPS" =~ ^[0-9]+(\.[0-9]+)?$ ]]
 then
-        HEIGHT=$4
+	echo "ERROR: Invalid FPS input"
```
Comment (jim-wang-intel) [inline benchmark-scripts/format_avc_mp4.sh:58]:
```suggestion
	echo "ERROR: FPS should be number"
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+if [ $statusCode==0 ]
+then
+    if [ -f "$DEFAULT_FILE_PATH_NAME" ]
+    then
+        echo "test PASSED: $DEFAULT_FILE_PATH_NAME has been downloaded."
+    else
+        echo "test FAILED: $DEFAULT_FILE_PATH_NAME has NOT been downloaded."
+    fi
+else
+    echo "test FAILED: $DEFAULT_FILE_PATH_NAME has NOT been downloaded."
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:74]:
```suggestion
    echo "test FAILED: expecting status code 0 but found $statusCode for $FILENAME_DOWNLOAD with FILE_URL_TO_DOWNLOAD."
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+echo
+echo "# test case 2: test with image, got statusCode 0 and test media file downloaded (happy path)"
+output=$(./format_avc_mp4.sh $FILENAME_DOWNLOAD $FILE_URL_TO_DOWNLOAD)  
+statusCode=$?
+echo "$statusCode"
+if [ $statusCode==0 ]
+then
+    if [ -f "$DEFAULT_FILE_PATH_NAME" ]
+    then
+        echo "test PASSED: $DEFAULT_FILE_PATH_NAME has been downloaded."
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:69]:
```suggestion
        echo "test PASSED: $DEFAULT_FILE_PATH_NAME has been downloaded for $FILENAME_DOWNLOAD with $FILE_URL_TO_DOWNLOAD."
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    else
+        echo "test FAILED: $DEFAULT_FILE_PATH_NAME has NOT been downloaded."
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:71]:
```suggestion
        echo "test FAILED: $DEFAULT_FILE_PATH_NAME has NOT been downloaded for $FILENAME_DOWNLOAD with $FILE_URL_TO_DOWNLOAD."
```

**Review by brian-intel — APPROVED** [review]

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+                echo "test FAILED: Second time download was missing skipped message!"
+            fi
+        else
+            echo "test FAILED: Second time download ERROR: $output."
+        fi
+    else
+        echo "test FAILED: download $FILENAME_DOWNLOAD first time failed"
+    fi
+else
+    echo "test FAILED: download $FILENAME_DOWNLOAD first time failed: $output."
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:105]:
```suggestion
    echo "test FAILED: download $FILENAME_DOWNLOAD first time failed: $output: expecting status code 0 but found $statusCode"
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+        if [ $statusCode==0 ]
+        then
+            if [[ "$output" == "$SUB"* ]]
+            then
+                echo "test PASSED: Second time download was skipped!"
+            else
+                echo "test FAILED: Second time download was missing skipped message!"
+            fi
+        else
+            echo "test FAILED: Second time download ERROR: $output."
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:99]:
```suggestion
            echo "test FAILED: Second time download ERROR: $output: expecting status code 0 but found $statusCode"
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+FPS=10
+output=$(./format_avc_mp4.sh $FILENAME_DOWNLOAD $FILE_URL_TO_DOWNLOAD $WIDTH $HEIGHT $FPS)  
+statusCode=$?
+FILE_PATH_NAME=../sample-media/test/test-$WIDTH-$FPS-bench.mp4
+if [ $statusCode==0 ]
+then
+    if test -f "$FILE_PATH_NAME"; then
+        echo "test PASSED: with input width $WIDTH, height $HEIGHT, FPS $FPS."
+    else
+        echo "test FAILED: with input width $WIDTH, height $HEIGHT, FPS $FPS."
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:124]:
```suggestion
        echo "test FAILED: with input width $WIDTH, height $HEIGHT, FPS $FPS: couldn't find downloaded file $FILE_PATH_NAME"
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    fi
+else
+    echo "test FAILED: with input width $WIDTH, height $HEIGHT, FPS $FPS."
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:127]:
```suggestion
    echo "test FAILED: with input width $WIDTH, height $HEIGHT, FPS $FPS: expecting status code 0 but found $statusCode"
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+fi
+cleanupTestFolderContent
+
+# test case 5: input Width should be integer type
+echo
+echo "# test case 5: input Width should be integer type"
+WIDTH=8abv
+HEIGHT=720
+FPS=10
+EXPECTED_MESSAGE="ERROR: Invalid width input"
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:137]:
```suggestion
EXPECTED_MESSAGE="ERROR: width should be integer"
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+fi
+cleanupTestFolderContent
+
+# test case 6: input Height should be integer type
+echo
+echo "# test case 6: input Height should be integer type"
+WIDTH=1035
+HEIGHT=8.0fd
+FPS=10
+EXPECTED_MESSAGE="ERROR: Invalid height input"
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:159]:
```suggestion
EXPECTED_MESSAGE="ERROR: height should be integer"
```

File: benchmark-scripts/test_format_avc_mp4.sh [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+fi
+cleanupTestFolderContent
+
+# test case 7: input FPS should be float or integer type
+echo
+echo "# test case 7: input FPS should be float or integer type"
+WIDTH=1035
+HEIGHT=335
+FPS=a.09
+EXPECTED_MESSAGE="ERROR: Invalid FPS input"
```
Comment (jim-wang-intel) [inline benchmark-scripts/test_format_avc_mp4.sh:181]:
```suggestion
EXPECTED_MESSAGE="ERROR: FPS should be number"
```

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

**Review by jstein13 — APPROVED** [review]

**Pull request merged**